src/modal/internal.tsx

```diff
@@ -95,6 +96,10 @@ export default function InternalModal({
     }
   };
 
+  // We use an empty div element at the end of the content slot as a sentinel
+  // to detect when the user has scrolled to be bottom.
+  const { ref: stickySentinelRef, isIntersecting: footerStuck } = useIntersectionObserver();
```

Comment:
Sadly there is no native way to detect if a sticky element is stickied or not. Intersection Observer seems to be the most common pattern to work around this, and it seems to work quite well. Browser support also looks good: https://caniuse.com/intersectionobserver.